controllers/spacebindingcleanup/spacebinding_cleanup_controller.go

```diff
@@ -89,21 +90,37 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound Space")
 	}
 
+	// ensure the MUR exists
+	requeueAfter, err := r.ensureMURExists(ctx, spaceBinding)
+	return ctrl.Result{RequeueAfter: requeueAfter}, err
+}
+
+func (r *Reconciler) ensureMURExists(ctx context.Context, spaceBinding *toolchainv1alpha1.SpaceBinding) (time.Duration, error) {
+	logger := log.FromContext(ctx).WithName("cleanup")
+
+	// if publicViewer is enabled and the SpaceBinding is related to the PublicViewer,
+	// do not check for MUR existence
+	cfg, err := toolchainconfig.GetToolchainConfig(r.Client)
+	if err != nil {
+		return norequeue, errs.Wrapf(err, "unable to get toolchainconfig")
+	}
+	if cfg.PublicViewer().Enabled() && spaceBinding.Spec.MasterUserRecord == toolchainv1alpha1.KubesawAuthenticatedUsername {
+		return norequeue, nil
```

Comment:
@filariow just for my understanding, why do we not want the reconciler to reque in the above scenario.?

Reply:
because in this scenario the existance of the SpaceBinding is fine even if the MUR does not exist. The `kubesaw-authenticated` is a special one that has no MUR.
Hence, returning `norequeue` here we are skipping the SpaceBinding deletion for this special user